app/src/main/java/com/kickstarter/viewmodels/usecases/GetShippingRulesUseCase.kt

```diff
@@ -233,10 +241,23 @@ class GetShippingRulesUseCase(
     ): ShippingRule =
         if (project.isBacking()) ShippingRule.builder()
             .apply {
+                val backing = project.backing()
                 val locationId = project.backing()?.locationId() ?: 0L
-                val locationName = project.backing()?.locationName() ?: ""
-
-                this.location(Location.Builder().id(locationId).name(locationName).displayableName(locationName).build())
+                this.id(locationId)
+                val reward = backing?.reward()?.let {
+                    if (RewardUtils.shipsToRestrictedLocations(it)) {
```

Comment:
for restricted locations reward will have the list of shipping rules with proper location id, name and cost